### DISCUSSION TITLE
Cannot find module '#app' when importing configuration from plugin file

### DISCUSSION THREAD

**Discussion by dargmuesli** [nuxt/nuxt#17174] [Questions] [answered]
When I import something from a plugin file in my `nuxt.config.ts`, that imports e.g. `defineNuxtPlugin` from `#app` in turn, I receive the error `Cannot find module '#app'`, even for `nuxi prepare` only.

In my case, I want to reuse a constant, that is provided via a plugin to the app, in the configuration too. It seems like I have to extract those exports from the plugin file to a separate file as a workaround.

Is this a bug or a won't fix? I'm unsure if it's an anti-pattern to import configuration from a plugin file.

Here is :heart: for minimal reproductions: https://codesandbox.io/s/cranky-margulis-piykwf?file=/plugins/util.ts

**Comment by jhisham** [comment]
I believe the issue is with explicitly importing files from your `plugin` directory. In essence, you don't really have to with Nuxt 3. This is because it is already auto-imported by Nuxt.

Per the documents:
>  Nuxt auto-imports helper functions, composables and Vue APIs to use across your application without explicitly importing them. Based on the directory structure, every Nuxt application can also use auto-imports for its own components, composables and plugins. Components, composables or plugins can use these functions.

Since you've already provided a helper function, you can call it from another file like so:

```
const $isComponentEnabled = useNuxtApp()
```

If for some reason you'd still like to use explixit imports, per the documents [Explicit imports](https://v3.nuxtjs.org/guide/concepts/auto-imports#explicit-imports) you can use the #imports alias:

```
import { isComponentEnabled } from '#imports'
```

**Reply by dargmuesli** [reply]
I wanted to use a function of a plugin in the configuration itself. I didn't try to import a plugin to make it available through the config to the rest of the app.
As Daniel pointed out below, it's simply not meant to be done this way. That's what I was wondering about.
I decided to ask here, because the error message did not lead me to the actual issue right away and after quite some time of debugging I found this cause. If someone else gets the error too, they now have a pointer to a possible cause on the internet at least ;)

**Comment by danielroe** [comment, marked as answer]
Within your `nuxt.config` context you do not have access to any aliases like `#app`. So you should not import from your runtime app into your config file. (And vice versa.)
